src/System.Numerics.Matrices/src/System/Numerics/IMatrix.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace System.Numerics.Matrices
+{
+    interface IMatrix
+    {
+        double this[int col, int row] { get; set; }
```

Comment:
Yeah, I realize double is a reasonable choice given how our type system and CLR work today. I would be interested in exploring what would we need to do in the type system and the CLR to enable scenarios like such generic matrix type. 
We explored many solutions here in the past, but never pulled the trigger. Adding @CarolEidt who worked in this space int the past.